### PR TITLE
feat: show longer dbt descriptions for fields, including markdown support

### PR DESCRIPTION
**Pull request by filp**
Closes: #1174 

### Description:

Adds an improved UX for viewing dbt field descriptions:

Screenshot 2024-02-20 at 22 15 58
Screenshot 2024-02-20 at 22 16 17
Screenshot 2024-02-20 at 22 16 25
Screenshot 2024-02-20 at 22 17 16
Screenshot 2024-02-20 at 22 17 09

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by hamzahc1** [comment]
@filp what's left to do to open this PR for review?

**Comment by filp** [comment]
@hamzahc1 there was a quirk with rendering modals from within the field tree on the left that I had to rework, and some design feedback to integrate. Opened it up for review now!

**Comment by filp** [comment]
Currently still includes one quirk I couldn't find a satisfactory solution for: opening the description modal from the hovercard doesn't immediately close the hovercard even though the modal is already visible, because the user's hover state is still keeping the card open.

Will take another stab at a solution, but IMO doesn't block reviewing the overall change.

File: packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeSingleNode.tsx [hunk 1]
```diff
@@ -10,23 +10,128 @@ import {
...
+    const lines = description.split('\n');
+
+    /**
+     * Collect up to 5 description lines, breaking early if we find anything that
+     * looks like a code block or a table.
+     */
+    const truncatedLines = lines
+        .slice(0, 4)
+        .reduce<string[]>((truncated, line, idx) => {
+            if (idx === 0 || (line.at(0) !== '|' && !line.startsWith('```'))) {
```
Comment (rephus) [inline packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeSingleNode.tsx:60]: so if the description starts with a block of code we will not show anything, right ?  I guess it kind of makes sense...

**Review by rephus — APPROVED** [review]
Tested and review, it look good to me, nice job!

Tip for make reviews easier: it would be nice if you can include a `before` and `after` screenshot with the change. 

I can see `last name` on render has a long description, but I don't see this change on jaffle shop in this PR. I think it would be nice to push that to the repo as well (or a shorter version), so we have something to test this again in the future.  

> Currently still includes one quirk I couldn't find a satisfactory solution for: opening the description modal from the hovercard doesn't immediately close the hovercard even though the modal is already visible, because the user's hover state is still keeping the card open.

I also noticed this, if you can fix it great, but if not, I think it would be ok to ship it as it is.

**Comment by filp** [comment]
Thank you @rephus! I made a couple of corrections based on your feedback:

- Tweaked the description for 'Order status' to include a bit more content to test this feature.
- Fixed a pending issue with markdown headings auto-linking
- Fixed the weird interaction between the hover-card and the modal by using a controlled Popover instead
- Simplified/improved the on-hover preview by rendering the full description as markdown, and instead clamping the visible amount in the hover card (so there's less magic when deciding what to render).

See a quick gif here:

Kapture 2024-02-23 at 14 24 01

Would love it if you could take another quick look and share you opinion!

**Review by rephus — APPROVED** [review]
Everything look good, it looks even better now. This hasn't changed though. Not sure if you are planning to do it 

https://github.com/lightdash/lightdash/pull/9055#issuecomment-1961214743

**Review by rephus — APPROVED** [review]

**Comment by filp** [comment]
Checked the failing e2e test locally, doesn't appear to be related.

**Pull request merged**